boot/cmdline.go

```diff
@@ -124,60 +124,67 @@ func getBootloaderManagingItsAssets(where string, opts *bootloader.Options) (boo
 		// the bootloader cannot manage its scripts
 		return nil, errBootConfigNotManaged
 	}
-	managed, err := mbl.IsCurrentlyManaged()
-	if err != nil {
-		return nil, err
-	}
-	if !managed {
-		return nil, errBootConfigNotManaged
-	}
 	return mbl, nil
 }
 
-// ComposeRecoveryCommandLine composes the kernel command line used when booting
-// a given recovery mode system.
-func ComposeRecoveryCommandLine(model *asserts.Model, system string) (string, error) {
+const (
+	currentEdition = iota
+	candidateEdition
+)
+
+func composeCommandLine(model *asserts.Model, currentOrCandidate int, mode, system string) (string, error) {
 	if model.Grade() == asserts.ModelGradeUnset {
 		return "", nil
 	}
-
+	if mode != ModeRun && mode != ModeRecover {
+		return "", fmt.Errorf("internal error: unsupported command line mode %q", mode)
+	}
 	// get the ubuntu-seed bootloader
```

Comment:
this is comment is a bit confusing, we will get either the boot one and the seed one in the if below